src/index.ts

```diff
@@ -1627,61 +1648,67 @@ const configShardusEndpoints = (): void => {
-          shardusAddress = toShardusAddress(address, AccountType.Account)
-        }
-        const hexBlockNumber = req.query.blockNumber as string
-        const hexBlockNumberStr = isHexString(hexBlockNumber) ? hexBlockNumber : null
+    if (!req.query.type) {
+      let shardusAddress = address.toLowerCase()
+      if (address.length === 42) {
+        shardusAddress = toShardusAddress(address, AccountType.Account)
+      }
+      const hexBlockNumber = req.query.blockNumber
+      const hexBlockNumberStr = isHexString(hexBlockNumber) ? hexBlockNumber : null
 
-        let data: WrappedEVMAccount
-        if (isArchiverMode() && hexBlockNumberStr) {
-          data = await AccountsStorage.fetchAccountDataFromCollector(shardusAddress, hexBlockNumberStr)
-          if (!data) {
-            return res.json({ account: null })
-          }
-        } else {
-          const account = await shardus.getLocalOrRemoteAccount(shardusAddress)
-          if (!account) {
-            return res.json({ account: null })
-          }
-          data = account.data as WrappedEVMAccount
+      let data
+      if (isArchiverMode() && hexBlockNumberStr) {
+        data = await AccountsStorage.fetchAccountDataFromCollector(shardusAddress, hexBlockNumberStr)
+        if (!data) {
+          return { account: null }
         }
-        fixDeserializedWrappedEVMAccount(data)
-        const readableAccount = await getReadableAccountInfo(data)
-        if (readableAccount) return res.json({ account: readableAccount })
-        else res.json({ account: data })
       } else {
-        let accountType: number
-        if (typeof req.query.type === 'string') accountType = parseInt(req.query.type)
-        if (AccountType[accountType] == null) {
-          return res.json({ error: 'Invalid account type' })
-        }
-        const secondaryAddressStr: string = (req.query.secondaryAddress as string) || ''
-        if (secondaryAddressStr !== '' && secondaryAddressStr.length !== 66) {
-          return res.json({ error: 'Invalid secondary address' })
-        }
-        const shardusAddress = toShardusAddressWithKey(address, secondaryAddressStr, accountType)
         const account = await shardus.getLocalOrRemoteAccount(shardusAddress)
-        const readableAccount = convertBigIntsToHex(account)
-        return res.json({ account: readableAccount })
+        if (!account) {
+          return { account: null }
+        }
+        data = account.data
       }
-    } catch (error) {
-      res.json({ error })
+
+      fixDeserializedWrappedEVMAccount(data)
+      const readableAccount = await getReadableAccountInfo(data)
+      return readableAccount ? { account: readableAccount } : { account: data }
+    } else {
+      let accountType = parseInt(req.query.type)
+      if (AccountType[accountType] == null) {
+        return { error: 'Invalid account type' }
+      }
+
+      const secondaryAddressStr = (req.query.secondaryAddress || '').toString()
```

Comment:
QQ since there is a length validation: are we sanitising inputs? I mean for this one what if we get string with 66 spaces. How does it impact the flow?

Reply:
It will just fail to get the account later one.  No issue.